### PR TITLE
Add length limitation to name and address fields

### PR DESCRIPTION
**Pull request by ostapneko**
@davidrapson Should I also do the front end validations or are yo uhappy to do them?

### CODE REVIEW

**Comment by davidrapson** [comment]
@ostapneko If you are OK to update the validations on the client-side too that would be great.

**Comment by ostapneko** [comment]
@davidrapson Will do

**Comment by ostapneko** [comment]
@davidrapson I did my best on the front end side to make it feature complete without duplication, but I'll leave the polishing to you I guess as I comply to the JS coding standards.

File: assets/javascripts/modules/checkout/personalDetails.js [hunk 1]
```diff
@@ -13,25 +15,65 @@ define([
...
+        return {
+            el: containerEl,
+            maxLength: maxLength,
+            input: $('.js-input', containerEl)[0],
+            error: $('.js-error-message', containerEl)[0]
+        };
+    }
+
+    var firstNameField = new TextField(formEls.$FIRST_NAME_CONTAINER, MAX_NAME_LENGTH);
+    var lastNameField = new TextField(formEls.$LAST_NAME_CONTAINER, MAX_NAME_LENGTH);
```
Comment (davidrapson) [inline assets/javascripts/modules/checkout/personalDetails.js]: You don't need the `new` here.

File: assets/javascripts/modules/checkout/personalDetails.js [hunk 2]
```diff
@@ -13,25 +15,65 @@ define([
     var FIELDSET_COMPLETE = 'data-fieldset-complete';
     var HIDDEN_CLASS = 'is-hidden';
 
+    var MAX_NAME_LENGTH = 50;
+    var MAX_EMAIL_LENGTH = 240;
+    var MAX_ADDRESS_LENGTH = 255;
+
+    var ERROR_MESSAGES = {
+        FIELD_TOO_LONG: 'This field is too long',
```
Comment (davidrapson) [inline assets/javascripts/modules/checkout/personalDetails.js]: We should tell the user how long is too long

File: app/views/checkout/payment.scala.html [hunk 3]
```diff
@@ -83,23 +83,23 @@
...
                                 <label class="label" for="address-line-1">Address line 1</label>
                                 <input type="text" class="input-text js-input" id="address-line-1" name="personal.address.address1" value="@form.data.get("personal.address.address1")">
-                                @fragments.forms.errorMessage("This field is required")
+                                @fragments.forms.errorMessage("")
                             </div>
                             <div class="form-field js-checkout-street">
                                 <label class="label optional-marker" for="address-line-2">Address line 2</label>
                                 <input type="text" class="input-text js-input" id="address-line-2" name="personal.address.address2" value="@form.data.get("personal.address.address2")">
-                                @fragments.forms.errorMessage("This field is required")
+                                @fragments.forms.errorMessage("")
```
Comment (davidrapson) [inline app/views/checkout/payment.scala.html]: Is it worth still providing a default message, even if we override in most cases?

**Comment by davidrapson** [comment]
:+1: LGTM aside from a couple of comments.

**Comment by TesterSpike** [comment]
The error messages are not removed when the user replaces the overly long text with appropriately length text.  
I'll add my vote to improving the text of the error messages so that the user knows what the limits are.

**Comment by davidrapson** [comment]
As @TesterSpike says, on second look this doesn't actually validate fields correctly. The values are not live updated. Going to close this one and put up a new PR to resolve this.

**Pull request closed without merging**